pkg/api/filter/filter.go

```diff
@@ -0,0 +1,33 @@
+package filter
+
+import "net/http"
+
+// Filter is a wrapper around a http.HandlerFunc.
+type Filter interface {
+	Filter(http.HandlerFunc) http.HandlerFunc
+}
+
+type chain struct {
+	filters []Filter
+}
+
+// NewFilterChain returns a Filter capable of applying
+// the provided Filter(s) to another http.HandlerFunc
+func NewFilterChain(filterChain []Filter) Filter {
```

Comment:
Can we possibly change the signature to this:

```
func NewFilterChain(filters ...Filter) Filter
```

That will let callers off the hook for explicitly constructing a slice that they likely do not need otherwise.

Reply:
Also, maybe s/NewFilterChain/NewChain/? "filter" is already specified by the package name. `filter.NewChain(...)` reads nice and clean.

Reply:
Yep!